unlock-app/src/utils/types.ts

```diff
@@ -0,0 +1,23 @@
+import { TransactionType } from '../unlockTypes'
+
+/**
+ * Mapping function to change the types received from unlock-js into types
+ * used by unlock-app
+ */
+export const transactionTypeMapping = (type: string) => {
+  if (type === 'LOCK_CREATION') {
+    return TransactionType.LOCK_CREATION
+  } else if (type === 'KEY_PURCHASE') {
+    return TransactionType.KEY_PURCHASE
+  } else if (type === 'WITHDRAWAL') {
+    return TransactionType.WITHDRAWAL
+  } else if (type === 'UPDATE_KEY_PRICE') {
+    return TransactionType.UPDATE_KEY_PRICE
+  } else {
+    // Not sure but that should not happen
```

Comment:
let's throw here?

Reply:
Welll, that's what I had initially but then remembered this is called inside the middleware... where we do not want to throw, right?

Reply:
Maybe we can wrap in try/catch and dispatch setError, but we definitely want to catch this edge case somehow?

Reply:
Since this should never happen I am actually in favor of keeping things simple here for now.

Reply:
How about we console.log() here? If it does happen, we would have a devil of a time tracking the bug down otherwise :)

Reply:
or better yet, can we use an enum for the parameter? maybe static typing will help us catch it?